### ISSUE TITLE
nodejs oncluster builds using s2i fails to deploy image when build envs is empty

### ISSUE THREAD

**Issue by jrangelramos** [knative/func#1363]
When building a nodejs on cluster using s2i builder, the function fails to deploy

```
$ func deploy --registry ttl.sh/knfuncw0eie3j8 --remote --builder s2i
   Creating Pipeline resources
🕓 Running Pipeline: Deploying function to the cluster
Error: failed to run pipeline: function pipeline run has failed with message: 

Deploying image 'ttl.sh/knfuncw0eie3j8/node-s2i:latest' with digest 'sha256:a765e4ac088fd6c2619338a940431cc70c43f46ceec0c2b2a3b15cbe90b41568'. Build and push are disabled.
⬆️  Deploying function to the cluster
Error: timeout: service 'node-s2i' not ready after 120 seconds

$ kubectl logs -f node-s2i-00003-deployment-7c9f9b58c-xshxj -c queue-proxy
{"severity":"INFO","timestamp":"2022-10-19T17:17:44.267040782Z","logger":"queueproxy","caller":"sharedmain/main.go:270","message":"Starting queue-proxy","commit":"812f1c8-dirty","knative.dev/key":"default/node-s2i-00003","knative.dev/pod":"node-s2i-00003-deployment-7c9f9b58c-xshxj"}
aggressive probe error (failed 202 times): Get "http://127.0.0.1:8080/health/readiness": dial tcp 127.0.0.1:8080: connect: connection refused

$ podman run -p 8080:8080 ttl.sh/knfuncw0eie3j8/node-s2i:latest
Trying to pull ttl.sh/knfuncw0eie3j8/node-s2i:latest...
Getting image source signatures
Copying blob 140da7f64cd2 done  
Copying blob b0d6828d61cd done  
Copying blob 1ded74a6469c done  
Copying blob 34d5ba0a419e done  
Copying blob c63dfb71651d done  
Copying blob 3c3681b59794 done  
Copying blob 86128da4bb5b done  
Copying config 29511e387e done  
Writing manifest to image destination
Storing signatures
Error: Env fields from image failed to parse: invalid environment variable: "="
```

It seems the fix applied on #1214 for empty build envs does not applies for s2i.

**Comment by zroubalik** [comment]
@jrangelramos this could be closed, right?

**Comment by jrangelramos** [comment]
yes! It is fixed by #1235
